src/main/java/com/netgrif/application/engine/petrinet/domain/arcs/reference/Reference.java

```diff
@@ -30,4 +30,11 @@ public int getMultiplicity() {
         }
         return multiplicity;
     }
+
+    public Reference clone() {
+        Reference clone = new Reference(this.reference, this.type);
+        clone.setReferencable(this.referencable == null ? null : this.referencable.clone());
```

Comment:
Does it work as intended? If you clone the referencable you create a new object and it does no longer reference an object from the cloned Petri net.

Reply:
Ok, now I understand, I will remove this method and callings for it.

Reply:
Corrected.